### PR TITLE
add a daemon thread doing regular maintenance

### PR DESCRIPTION
**Pull request by arnej27959**
* also, refactor test implementation

@bjorncs please review
@bjormel  @yngveaasheim FYI

### CODE REVIEW

File: logserver/src/test/java/com/yahoo/logserver/handlers/archive/ArchiverHandlerTestCase.java [hunk 1]
```diff
@@ -203,13 +203,20 @@ public void testRotation() throws IOException {
...
         String d = "target/tmp/logarchive";
         FilesArchived archive = new FilesArchived(new File(d));
         for (int i = 0; i < cache.maxEntries + 10; i++) {
             cache.put(i, new LogWriter(d+"/2018/12/31/17", 5, archive));
         }
         assertEquals(cache.size(), cache.maxEntries);
+
+        } catch (NullPointerException e) {
+            System.err.println("null: "+e);
+            e.printStackTrace();
```
Comment (bjorncs) [inline logserver/src/test/java/com/yahoo/logserver/handlers/archive/ArchiverHandlerTestCase.java]: This looks like debug code that was accidentally included in the commit.

File: logserver/src/main/java/com/yahoo/logserver/handlers/archive/FilesArchived.java [hunk 2]
```diff
@@ -37,19 +37,36 @@
...
+
     /**
-     * Creates an FilesArchive managing the given directory
+     * Creates an instance of FilesArchive managing the given directory
      */
     public FilesArchived(File rootDir) {
         this.root = rootDir;
-        maintenance();
+        rescan();
+        Thread thread = new Thread(this::run);
```
Comment (bjorncs) [inline logserver/src/main/java/com/yahoo/logserver/handlers/archive/FilesArchived.java]: Consider assigning a name to the thread (for easier investigating of stack dumps).

File: logserver/src/main/java/com/yahoo/logserver/handlers/archive/FilesArchived.java [hunk 3]
```diff
@@ -37,19 +37,36 @@
     private long maxAgeDays = 30; // GDPR rules: max 30 days
     private long sizeLimit = 30L * (1L << 30); // 30 GB
 
+    private void run() {
+        try {
+            Thread.sleep(125000); // 2 m 5 s
+            while (true) {
+                while (maintenance()) {
```
Comment (bjorncs) [inline logserver/src/main/java/com/yahoo/logserver/handlers/archive/FilesArchived.java]: The maintenance thread will stop if `maintenance()` throws an unchecked exception.

**Review by bjorncs — COMMENTED** [review]
See comments

File: logserver/src/test/java/com/yahoo/logserver/handlers/archive/ArchiverHandlerTestCase.java (hunk 1 above)
Comment (arnej27959) [reply logserver/src/test/java/com/yahoo/logserver/handlers/archive/ArchiverHandlerTestCase.java]: debug code removed.

File: logserver/src/main/java/com/yahoo/logserver/handlers/archive/FilesArchived.java (hunk 3 above)
Comment (arnej27959) [reply logserver/src/main/java/com/yahoo/logserver/handlers/archive/FilesArchived.java]: yes.

**Comment by arnej27959** [comment]
@bjorncs please take another look

**Review by bjorncs — APPROVED** [review]

**Pull request merged**